src/bootstrap/src/lib.rs

```diff
@@ -1950,6 +1950,20 @@ impl Build {
         t!(fs::remove_dir_all(dir))
     }
 
+    /// Make sure that `dir` will be an empty existing directory after this function ends.
+    /// If it existed before, it will be first deleted.
+    fn clear_dir(&self, dir: &Path) {
```

Comment:
Question: can this use `build_helper::remove_and_create_dir_all`?

Reply:
I would rather introduce the I/O helpers into bootstrap directly, so that they can integrate with (I/O) tracing spans.